driver-core/src/main/com/mongodb/internal/client/model/bulk/ConcreteClientUpdateOneModel.java

```diff
@@ -15,26 +15,32 @@
  */
 package com.mongodb.internal.client.model.bulk;
 
-import com.mongodb.client.model.bulk.ClientUpdateOptions;
+import com.mongodb.assertions.Assertions;
+import com.mongodb.client.model.bulk.ClientUpdateOneOptions;
 import com.mongodb.lang.Nullable;
 import org.bson.conversions.Bson;
 
 /**
  * This class is not part of the public API and may be removed or changed at any time.
  */
-public final class ConcreteClientUpdateOneModel extends AbstractClientUpdateModel implements ClientWriteModel {
+public final class ConcreteClientUpdateOneModel extends AbstractClientUpdateModel<ConcreteClientUpdateOneOptions> implements ClientWriteModel {
     public ConcreteClientUpdateOneModel(
             final Bson filter,
             @Nullable
             final Bson update,
             @Nullable
             final Iterable<? extends Bson> updatePipeline,
-            @Nullable final ClientUpdateOptions options) {
-        super(filter, update, updatePipeline, options);
+            @Nullable final ClientUpdateOneOptions options) {
+        super(filter, update, updatePipeline,
+                options == null ? ConcreteClientUpdateOneOptions.MUTABLE_EMPTY : (ConcreteClientUpdateOneOptions) options);
     }
 
     @Override
-    String getToStringDescription() {
-        return "ClientUpdateOneModel";
+    public String toString() {
+        return "ClientUpdateOneModel{"
+                + "filter=" + getFilter()
+                + ", update=" + (getUpdate().isPresent() ? getUpdate() : getUpdatePipeline().orElseThrow(Assertions::fail))
```

Comment:
The same thing I suggested in `ConcreteClientUpdateManyModel`:

```suggestion
                + ", update=" + getUpdate().map(Object::toString).orElseGet(() ->
                        getUpdatePipeline().map(Object::toString).orElseThrow(Assertions::fail))
```